commands/bmc

```diff
@@ -0,0 +1,252 @@
+    date --date "@${sec}" --utc +'%a %d %b %Y %T UTC'
+  else
+    date --date "@${sec}" +'%a %d %b %Y %T UTC%z'
+  fi
+}
+
+# @doc cmd_info
+# General BMC information
+function cmd_info {
+  subcommand "$@"
+}
+
+# @doc cmd_info_uptime
+# Show BMC uptime
+#   -c, --compact   Show last boot time in compact format
+#   -u, --utc       Show last boot time in UTC timezone
+function cmd_info_uptime {
+  local compact=""
+  local utc=""
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      -c | --compact) compact="y";;
+      -u | --utc) utc="y";;
+      *) abort_badarg "$1";;
+    esac
+    shift
+  done
+
+  local sec="$(awk -F. '{print $1}' /proc/uptime)"
+  local up="$(($(date +%s) - sec))"
+
+  printf 'Uptime is %dd %dh %dm %ds.\n' $((sec / 86400)) \
+                                        $((sec % 86400 / 3600)) \
+                                        $((sec % 3600 / 60)) \
+                                        $((sec % 60))
+  echo -n "Last BMC start was on "
+  format_date "${compact}" "${utc}" "${up}"
+}
+
+# @sudo cmd_info_version admin,operator,user
+# @doc cmd_info_version
+# Show version of BMC and BIOS/UEFI
+#   -b, --bmc   Show BMC version
+#   -u, --bios  Show BIOS/UEFI version
+function cmd_info_version {
+  local bmc=""
+  local bios=""
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      -b | --bmc) bmc="y";;
+      -u | --bios) bios="y";;
+      *) abort_badarg "$1";;
+    esac
+    shift
+  done
+
+  if [[ -z "${bmc}${bios}" ]]; then
+    bmc="y"
+    bios="y"
+  fi
+
+  if [[ -n "${bmc}" ]]; then
+    fwupdate --version | grep "^BMC "
+  fi
+  if [[ -n "${bios}" ]]; then
+    fwupdate --version | grep "^Host "
+  fi
+}
+
+# @doc cmd_datetime
+# BMC date and time configuration
+function cmd_datetime {
+  subcommand "$@"
+}
+
+# @sudo cmd_datetime_set admin
+# @doc cmd_datetime_set
+# Set system time manually
+#   YYYY-MM-DD-HH-mm-SS   Time to set
+function cmd_datetime_set {
+  if [[ $# -eq 0 ]] || [[ $1 =~ ^\d{4}-\d\d-\d\d-\d\d-\d\d-\d\d$ ]]; then
+    echo "Please enter date and time in YYYY-MM-DD-HH-mm-SS format" >&2
+    return 1
+  fi
+
+  local new_date="${1:0:10}"
+  local new_time="${1:11}"
+  date "${new_date} ${new_time//-/:}"
+
+  echo "System time is updated."
+}
+
+# @doc cmd_datetime_show
+# Show system uptime
+#   -c, --compact   Show last boot time in compact format
+#   -u, --utc       Show last boot time in UTC timezone
+function cmd_datetime_show {
+  local compact=""
+  local utc=""
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      -c | --compact) compact="y";;
+      -u | --utc) utc="y";;
+      *) abort_badarg "$1";;
+    esac
+    shift
+  done
+  format_date "${compact}" "${utc}" "$(date +%s)"
+}
+
+# @sudo cmd_datetime_ntpconfig admin
+# @doc cmd_datetime_ntpconfig
+# Configure NTP server connection
+function cmd_datetime_ntpconfig {
+  netconfig "$@"
+}
+
+# @sudo cmd_ifconfig admin
+# @doc cmd_ifconfig
+# BMC network configuration
+function cmd_ifconfig {
+  netconfig "$@"
```

Comment:
And here too

Reply:
When function doesn't contain help - all parameters are passed to netconfig.

Reply:
Yes, these utils are in the /sbin directory and an end user doesn't have it in the PATH.